src/measurement/feature_directional_statistics.h

```diff
@@ -60,7 +60,13 @@ class FeatureDirectionalStatistics : public LineBased {
                   }
                }
                if( data ) {
-                  data->Push( *grey );
+                  const auto x = *grey;
+                  auto &p = lut_[x];
```

Comment:
How does this fare  when run across multiple threads? `std::unordered_map` is not atomic, is it?
I’m thinking we need one lookup table for each thread.

Reply:
oh yes definitely, I misunderstood the API - I thought there would be one of these classes instantiated per thread already.

Reply:
I was very wrong here, there is no multithreading in the measurement infrastructure.